lz4/stream/_stream.c

```diff
@@ -0,0 +1,1404 @@
+
+static inline int
+lz4_store_block_length (int block_length, int block_length_size, char * buf)
+{
+  int status = 1;
+
+  switch (block_length_size)
+    {
+      case 0: /* do nothing */
+        break;
+
+      case 1:
+        {
+          if (block_length > UINT8_MAX)
+            {
+              status = 0;
+              break;
+            }
+          store_le8 (buf, (uint8_t) (block_length & UINT8_MAX));
+        }
+        break;
+
+      case 2:
+        {
+          if (block_length > UINT16_MAX)
+            {
+              status = 0;
+              break;
+            }
+          store_le16 (buf, (uint16_t) (block_length & UINT16_MAX));
+        }
+        break;
+
+      case 4:
+        {
+          if (block_length > UINT32_MAX)
+            {
+              status = 0;
+              break;
+            }
+          store_le32 (buf, (uint32_t) (block_length & UINT32_MAX));
+        }
+        break;
+
+      default: /* unsupported cases */
+        status = 0;
+        break;
+    }
+
+  if (status != 1)
+    {
+      PyErr_SetString (LZ4StreamError, "Compressed stream size too large");
+    }
+
+  return status;
+}
+
+static PyObject *
+_compress_bound (PyObject * Py_UNUSED (self), PyObject * args, PyObject * kwargs)
+{
+  PyObject * py_dest = NULL;
+  uint32_t   input_size;
+
+  /* Positional arguments: input_size
+   * Keyword arguments   : none
+   */
+#if IS_PY3
+#define _ARG_FMT  "I"
+#else
+#define _ARG_FMT  "I"
+#endif
+  if (!PyArg_ParseTuple (args, _ARG_FMT,
+                         &input_size))
+    {
+      goto exit_now;
+    }
+#undef _ARG_FMT
+
+  py_dest = PyLong_FromUnsignedLong(_LZ4_compressBound(input_size));
+
+  if (py_dest == NULL)
+    {
+      PyErr_NoMemory ();
+    }
+
+exit_now:
+  return py_dest;
+}
+
+static PyObject *
+_input_bound (PyObject * Py_UNUSED (self), PyObject * args, PyObject * kwargs)
+{
+  PyObject * py_dest = NULL;
+  uint32_t   compress_max_size;
+
+  /* Positional arguments: compress_max_size
+   * Keyword arguments   : none
+   */
+#if IS_PY3
+#define _ARG_FMT  "I"
+#else
+#define _ARG_FMT  "I"
+#endif
+  if (!PyArg_ParseTuple (args, _ARG_FMT,
+                         &compress_max_size))
+    {
+      goto exit_now;
+    }
+#undef _ARG_FMT
+
+  py_dest = PyLong_FromUnsignedLong(_LZ4_inputBound(compress_max_size));
+
+  if (py_dest == NULL)
+    {
+      PyErr_NoMemory ();
+    }
+
+exit_now:
+  return py_dest;
+}
+
+static inline int
+lz4_compress_generic (_LZ4StreamContext_t * lz4_ctxt, char * source, int source_size,
+                      char * dest, int dest_size)
+{
+  int comp_len;
+
+  if (lz4_ctxt->config.comp == HIGH_COMPRESSION)
+    {
+      LZ4_streamHC_t * lz4_state = lz4_ctxt->lz4_state.compress.hc;
+
+      comp_len = LZ4_compress_HC_continue (lz4_state, source, dest, source_size, dest_size);
+    }
+  else
+    {
+      LZ4_stream_t * lz4_state    = lz4_ctxt->lz4_state.compress.fast;
+      int            acceleration = (lz4_ctxt->config.comp != FAST)
```

Comment:
Remove extraneous whitespace. Please check the whole file and remove these alignment whitespaces - they're too numerous to comment on each one.